### PR TITLE
k-filter-textbox

### PR DESCRIPTION
**Pull request by christianmemije**
# Summary

* Add `k-filter-textbox`

## TODO

- [ ] Update guidelines. 

127 0 0 1-8000-style_guide

### CODE REVIEW

**Comment by indirectlylit** [comment]
formatting issue on smaller monitors in the user list page:

image

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  .k-filter
+    display: table
+    background-color: white
+    border: 1px solid $core-grey-300
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/k-filter-textbox/index.vue]: There are a few very slight styling issues:

* should have a 2px corner radius
* check sizing and icon placement
* font colors and sizes (particularly of the placeholder text) might be off?

image

----

[google doc](https://docs.google.com/document/d/1S5Z1w_-Yr4JwPrK8Lf26MV9WLXzUHyQBEMlAte_hFn8/edit#)

cc @khangmach

**Comment by indirectlylit** [comment]
not critical, but would be nice if the keyboard focus outline surrounded the entire component rather than just the inner textbox:

image

It's possible to make the input the full width and then use inner left and right padding to leave space for the icon and button.

**Comment by indirectlylit** [comment]
Looks like this issue might be related: https://github.com/learningequality/kolibri/issues/1476

Maybe worth taking a look at it in this context?

**Comment by christianmemije** [comment]
* Fixed the responsive issue by removing extra styling.
* Keyboard focus now surrounds entire box.

@indirectlylit

**Comment by indirectlylit** [comment]
This is looking good.

I still see some small formatting issues. It's hard for me to see the [mockups in the google doc](https://docs.google.com/document/d/1S5Z1w_-Yr4JwPrK8Lf26MV9WLXzUHyQBEMlAte_hFn8/edit) - @khangmach can you weigh in on fonts, spacing etc or provide the original redlined mockups?

This is how it currently looks in the PR:

image

In particular I believe the magnifying glass should have 16px padding on each side, and perhaps the font size/color might be different?

The one other issue I'm noticing is vertical alignment with other elements:

image

image

You can see the items on the right are too low.

**Comment by christianmemije** [comment]
Need redlines @khangmach :pray:

**Comment by khangmach** [comment]
@indirectlylit 

filter textbox

text color is our updated primary black: #212121 with an opacity of 0.5
font size in this mockup is 14px regular body text
textbox icon indeed does have a padding of 16px both sides

**Comment by christianmemije** [comment]
* Gave the icons `8px` right and left margins. `16px` was too much. 
* Made the icon use `$core-text-annotation` for color.
* Updated the `$core-text-annotation` value.

**Comment by christianmemije** [comment]
> The one other issue I'm noticing is vertical alignment with other elements
Due to the floats. :rage4: :rage4: :rage4: :rage4:

**Comment by indirectlylit** [comment]
> Due to the floats

Personally I always use `position: absolute; top: 0; right: 0` instead of floating and then give the left-aligned  content a `margin-right` to give it space.  Default floating behavior is often hard to predict and control.

**Comment by christianmemije** [comment]
> Personally I always use position: absolute; top: 0; right: 0 instead of floating 

Not responsive :/

**Comment by indirectlylit** [comment]
> Not responsive :/

Sure it is, when applied conditionally with `responsive-window` :)

**Comment by christianmemije** [comment]
Fixed alignment. Stuck with floats for now since there is still discussion of the placement/responsiveness of buttons and filters in tables that should be tackled separately.

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+<script>
+
+  import uiIcon from 'keen-ui/src/UiIcon';
+  import uiIconButton from 'keen-ui/src/UiIconButton';
+  /**
+   * Used to filter items via text input
+   */
+  export default {
+    name: 'k-filter-textbox',
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: `Kolibri: Name id "k-filter-textbox" should be in camelCase. Found in /Users/d/Projects/learning_equality/repos/kolibri/kolibri/core/assets/src/views/k-filter-textbox/index.vue`

Comment (christianmemije) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: But none of the component names are camelCase...

Comment (indirectlylit) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: Hm, not sure what that warning is referring to then

Comment (christianmemije) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: Let's merge and open and issue for these lint warnings?

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    padding-top: 0
+    padding-right: 40px
+    padding-bottom: 0
+    padding-left: 40px
+    width: calc(100% - 80px)
+    height: 40px
+    border: 1px solid $core-grey-300
+    border-radius: 2px
+    background-color: white
+    color: $core-text-default
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/k-filter-textbox/index.vue:120]: we should add `font-size: 14px` to this block.

currently it's not set, and shows at 16px:

image

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue (hunk 2 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: okay that's fine

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue (hunk 3 above)
Comment (christianmemije) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:120]: Added

File: kolibri/core/assets/src/views/k-filter-textbox/index.vue (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: These were the lint warnings you added to the extract_$trs plugin, @christianmemije.

Comment (rtibbles) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: (Now that we are using component names as message namespaces).

Comment (rtibbles) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: I assume that none of the other k-components have built in messages?

Comment (christianmemije) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: That makes sense.

Comment (christianmemije) [reply kolibri/core/assets/src/views/k-filter-textbox/index.vue:41]: Opened an issue regarding this: https://github.com/learningequality/kolibri/issues/2094

**Pull request merged**